### PR TITLE
Update solver_interface.jl

### LINKED ISSUE #12: length(rows) != length(cols) in add_linear_constraints!
This call

https://github.com/JuliaOpt/LinQuadOptInterface.jl/blob/master/src/constraints/scalaraffine.jl#L37

seems inconsistant with the interface contract

https://github.com/JuliaOpt/LinQuadOptInterface.jl/blob/master/src/solver_interface.jl#L153

### PR DESCRIPTION
**Pull request by IssamT**
Fixes #12

### CODE REVIEW

**Comment by odow** [comment]
I'm not sure enforcing the sorted order is a good idea.

It really needs to follow the [Gurobi docs](https://www.gurobi.com/documentation/7.5/refman/c_grbaddconstrs.html) since that is what we followed, or we should change the API to be more user-friendly (@joaquimg thoughts?)

**Comment by IssamT** [comment]
I think they are sorted in the classic CSR format, which is used by Gurobi. Am I wrong?

https://de.wikipedia.org/wiki/Compressed_Row_Storage

**Comment by odow** [comment]
I guess it's okay because we're calling it from LQOI. Can you add the example of how to call it from the issue please? That should help clear up any future confusion.

**Comment by joaquimg** [comment]
Does gurobi cplex and xpress require the sorted format?
I think for LP its critical to have efficient implementation for building problems.
If a single solver requires sorting, it could do sorting internally.

**Comment by joaquimg** [comment]
If all solver do need a sorted format I’d say it should be enforced.

File: src/solver_interface.jl [hunk 1]
```diff
@@ -149,8 +149,25 @@ function get_number_linear_constraints end
...
+
+To add the constraints:
+    1.0x + 2.0y <= 3.0
+            4.0y >= 5.0,
+the call is:
+    add_linear_constraints(m, 
+        [1, 3], 
+        [1, 2, 2], 
+        [1.0, 2.0, 4.0], 
+        ["<=", ">="],
```
Comment (mlubin) [inline src/solver_interface.jl]: These are strings, not `Cchar`s

File: src/solver_interface.jl [hunk 2]
```diff
@@ -149,8 +149,25 @@ function get_number_linear_constraints end
 
 Adds linear constraints of the form `Ax (sense) rhs` to the model `m`.
 
-The A matrix is given in triplet form `A[rows[i], cols[i]] = coef[i]` for all
-`i`, and `length(rows) == length(cols) == length(coefs)`.
+The A matrix is given in triplet form `A[row(i), cols[i]] = coef[i]` for all
+`i` where `row(i)` is the largest element in `rows` that is smaller or equal 
+than `i`. `rows` is a list of integers sorted in increasing order. It contains 
+the starting index (of `cols`) for each row. `length(cols) == length(coefs)` 
+and `length(rows) == length(sense) == length(rhs)`.
```
Comment (mlubin) [inline src/solver_interface.jl]: We should be moving away from interfaces that have lists of arrays of equal length when possible. Why not use a struct instead? Why are we redefining CSR matrices?

Comment (joaquimg) [reply src/solver_interface.jl]: Most solvers use CSR Why should they Need an extra conversion layer?

Comment (mlubin) [reply src/solver_interface.jl]: What about defining a struct for the CSR matrix then?

Comment (joaquimg) [reply src/solver_interface.jl]: That would be nice!

**Pull request closed without merging**